Source/1. Functional Programmingming/js/10_Lazy.js

```diff
@@ -0,0 +1,26 @@
+﻿//#10 Lazy
+var lazyFn = (function () {
+    "use strict";
+    return function (initialFn) {
+        if (typeof initialFn !== 'function') {
+            throw new TypeError(initialFn + ' is not a function');
+        }
+        var getLazyFn = function (args, result) {
+            return function () {
+                var compareArgsFn = function (curArg, index, array) { return curArg === args[index]; }
```

Comment:
I'd suggest writing the result of calculation into variable, to avoid this function being recalculate each time it is called